### ISSUE TITLE
Installed apps current locale available status

### ISSUE THREAD

**Issue by licaon-kter** [LonamiWebs/Stringlate#46]
Can this be detected? If so, it would be nice to signal if available or not, say below the checkmark: "XX" if available or "Add XX" if not ( where XX is current locale)

**Comment by XyLoNaMiyX** [comment]
> Can this be detected?

[Don't think so](https://developer.android.com/reference/android/content/pm/PackageInfo.html). The idea was good though, but I'm not going to dig into root access and do some fancy stuff over the `.apk` files only to see the locales it's available.

If anyone is up to look at how this could be done, well, help is appreciated.

**Comment by licaon-kter** [comment]
Root is an exceptional thing, yeah don't bother then.

**Comment by OffifialMITX** [comment]
You don't need root to access the apk, don't you (only if paid on Play Store)?

**Comment by XyLoNaMiyX** [comment]
> You don't need root to access the apk

You are right on that now that I think about it, APKShare doesn't need root. But still, I would need to extract the package somewhere and then scan it looking for the strings. It would be nice there was a way to simply tell the locales, but copying the .apk somewhere, extracting it and scanning it, is not quite fast.

**Comment by mquinson** [comment]
That could be displayed in the "discover repository" thing, don't you think?

**Comment by Lonami** [comment]
Yeah @mquinson the idea was to show a text on the same "application" item on the discover repositories thing, but the problem is I don't know how to check which locales are available in the first place :/

**Comment by mquinson** [comment]
Wouldn't it possible to make it a configuration item? As a user, that would definitely do the job.

**Comment by Lonami** [comment]
Sorry? Configuration item?

**Comment by mquinson** [comment]
Ok, sorry for my misunderstanding. I see 2 possible meaning of what you've said earlier:

(1) It is hard to find out which local the current user is prefering so you don't know what to display. In this case, just add a configuration item so that the user can tell you which local to use.

(2) It is hard to find out the current level of translation in each application of the list. In this case, the easiest is probably to do a partial checkout of the git tree, to find it out. I suspect that this is easier than messing with the apk file, but still not trivial.

**Comment by Lonami** [comment]
1. First use system default, then allow him to change it.
2. No, the idea is, since it's installed, I should have access to its resources, and so I should be able to tell which language is translated and which isn't, and simply show "this installed application you have doesn't have the locale you're using" or similar.
